### PR TITLE
adding integration with circle-ci

### PR DESCRIPTION
**Pull request by jkurz**
* building docker image on circle ci
* running unit tests in circle ci and not during npm phase
* running e2e tests in circle ci against running docker image
* pushing docker image to quay.io

### CODE REVIEW

**Comment by wilsonodk** [comment]
I have some issues with the approach that was taken for handling the tests.

**Comment by jkurz** [comment]
what would you suggest we do instead? we need to be able to bypass auth when we are doing E2E tests of the application?

**Comment by wilsonodk** [comment]
First, I'd use `npm test` to run the tests. 

Second, I'd add back the Chai dependency (since your tests depend on it). 

Third, I'd mock the call to auth instead of the odd bypassing. All of the tests done in the bash scripts, should be done using Mocha (with the Chai assertion library). You can then use Nock to mock all of the external calls.

**Comment by wilsonodk** [comment]
Also, it seems like there is a lot of cruft in the Circle YAML file to run the tests. 

Also, I'm a little confused on the container version in the YAML. Want to make sure we continue to use our customers version number when we register with CatalogIt after pushing to Quay.

**Comment by jkurz** [comment]
ok so I took out the bash tests that write to mongo. I agree those were not needed and we should be doing more testing with mocha, in terms of integrations. I have not added those tests in yet, as I just wanted to remove that stuff. I will add at a later time.

Also I wanted to make sure we were using the version from the package.json like you said, so now we are. We are also adding the branch name and build number to the tag, if the branch is not master. This is to match closer to what we have been doing.

File: set_build_values [hunk 1]
```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+have_jq=$(which jq)
+
+if [[ -z "${have_jq}" ]]; then
+  echo "You must have jq (https://stedolan.github.io/jq/) installed to use version-bump"
```
Comment (wilsonodk) [inline set_build_values]: `set_build_values` not `version-bump`

File: package.json [hunk 2]
```diff
@@ -8,7 +8,7 @@
   },
   "scripts": {
     "start": "node server.js",
-    "postinstall": "mocha tests;"
+    "test": "mocha tests"
```
Comment (wilsonodk) [inline package.json]: Probably shouldn't assume that mocha is globally installed.

**Comment by wilsonodk** [comment]
I'm still not a fan of having to to globally install Mocha and Chai. We work on a ton of different APIs/repos, that are all at different versions of those dependencies. For example, my currently globally installed Mocha is version `1.20.1`, but you require `^2.5.3`. So, there are breaking changes between the version I have, and the version this repo requires. Would be better to leverage npm's default behavior of locally installing dependencies, and then depending on those locally installed binaries. This insulates the repo from variances in local development and ensures that it works as expected in production environments.

**Comment by jkurz** [comment]
ok. I updated to just install with npm install and I am calling npm test in the circle build. Now everything is tested with local dependencies in the project.

**Pull request merged**